pkgs/by-name/in/install-grub-ng/src/builder/entries.rs

```diff
@@ -0,0 +1,401 @@
+            )?;
+        }
+
+        self.add_entry(
+            &format!(
+                "{name}{}{name_suffix}",
+                if !links.is_empty() { " - Default" } else { "" }
+            ),
+            path,
+            options,
+            current,
+        )?;
+
+        // Find all the children of the current default configuration
+        // Do not search for grand children
+        for link in &links {
+            // FIXME:
+            // This currently returns 1970-01-01 no matter what.
+            // These links seem to be have the correct creation date set, but not their
+            // modified date...?
+            let date = Self::generation_date_from_link(link)?;
+
+            let version = if let Ok(version) = fs::read_to_string(link.join("nixos-version")) {
+                version
+            } else {
+                let modules = link
+                    .join("kernel")
+                    .canonicalize()?
+                    .parent()
+                    .ok_or_else(|| {
+                        eyre!("Somehow {}/kernel doesn't have a parent..?", link.display())
+                    })?
+                    .join("lib/modules");
+
+                let Some(version) = fs::read_dir(&modules)?.find_map(|m| {
+                    m.ok()
+                        .and_then(|p| Some(p.path().file_name()?.to_string_lossy().into_owned()))
+                }) else {
+                    bail!("Could not deduce the current NixOS version")
+                };
+
+                version
+            };
+
+            let entry_name =
+                fs::read_to_string(link.join("configuration-name")).unwrap_or_else(|_| {
+                    format!(
+                        "({} - {date} - {version})",
+                        link.file_name().unwrap_or_default().to_string_lossy()
+                    )
+                });
+
+            self.add_entry(&format!("{name} - {entry_name}"), link, "", true)?;
+        }
+
+        if !current && !links.is_empty() {
+            writeln!(&mut self.inner, "}}")?;
+        }
+
+        Ok(())
+    }
+
+    fn generation_date_from_link(link: &Path) -> Result<time::Date> {
+        let sys_time = link.symlink_metadata()?.modified()?;
+
+        Ok(time::OffsetDateTime::from(sys_time).date())
+    }
+
+    fn add_entry(&mut self, name: &str, path: &Path, options: &str, current: bool) -> Result<()> {
+        let kernel_dir = path.join("kernel");
+        let initrd_dir = path.join("initrd");
+
+        if !(kernel_dir.exists() && initrd_dir.exists()) {
+            return Ok(());
+        }
+
+        let kernel_dir = self.copy_to_kernels_dir(&kernel_dir)?;
+        let initrd_dir = self.copy_to_kernels_dir(&initrd_dir)?;
+
+        // Include second initrd with secrets
+        let secrets_dir = self
+            .append_initrd_secrets(name, path, current)?
+            .unwrap_or_default();
+
+        // FIXME: $confName
```

Comment:
What does this mean? Maybe put more details in the comment.

Reply:
This is from the original script, and I have no idea what it means either

Reply:
It seems to date back to https://github.com/NixOS/nixpkgs/commit/f07f221f0ee46e64bb8a51e04dec6bade8645cfa. I say we just remove it. If it still makes sense, I'm sure we'll figure it out again at some point, no use keeping a 12 year old FIXME no one understands :stuck_out_tongue:

Reply:
Geez it's *that* old? ...Guess the Perl parts of Nixpkgs are the most ancient after all